core/templates/pages/teach-page/teach-page.component.ts

```diff
@@ -16,64 +16,65 @@
+@Component({
+  selector: 'teach-page',
+  templateUrl: './teach-page.component.html',
+  styleUrls: []
+})
+export class TeachPageComponent implements OnInit {
+  TAB_ID_TEACH: string = 'teach';
+  TAB_ID_PARTICIPATION: string = 'participation';
+  TEACH_FORM_URL: string = 'https://goo.gl/forms/0p3Axuw5tLjTfiri1';
+  ALLOWED_TABS: string[] = [this.TAB_ID_TEACH, this.TAB_ID_PARTICIPATION];
+  activeTabName: string = this.TAB_ID_TEACH;
 
-      const ALLOWED_TABS = [
-        ctrl.TAB_ID_TEACH, ctrl.TAB_ID_PARTICIPATION];
+  constructor(
+    private siteAnalyticsService: SiteAnalyticsService,
+    private urlInterpolationService: UrlInterpolationService,
+    private windowRef: WindowRef
+  ) {}
 
-      ctrl.activeTabName = ctrl.TAB_ID_TEACH;
-
-      ctrl.onTabClick = function(tabName) {
-        // ---- Update hash ----
-        WindowRef.nativeWindow.location.hash = '#' + tabName;
-        ctrl.activeTabName = tabName;
-      };
+  ngOnInit(): void {
+    const hash = this.windowRef.nativeWindow.location.hash.slice(1);
+    if (this.ALLOWED_TABS.includes(hash)) {
+      this.activeTabName = hash;
+    }
+    this.windowRef.nativeWindow.onhashchange = () => {
+      const hashChange = this.windowRef.nativeWindow.location.hash.slice(1);
+      if (this.ALLOWED_TABS.includes(hashChange)) {
+        this.activeTabName = hashChange;
+      }
+    };
+  }
 
-      ctrl.getStaticImageUrl = function(imagePath) {
-        return UrlInterpolationService.getStaticImageUrl(imagePath);
-      };
+  onTabClick(tabName: string): Window {
+    // ---- Update hash ----
+    this.windowRef.nativeWindow.location.hash = '#' + tabName;
+    this.activeTabName = tabName;
+    return this.windowRef.nativeWindow;
+  }
 
-      ctrl.onApplyToTeachWithOppia = function() {
-        SiteAnalyticsService.registerApplyToTeachWithOppiaEvent();
-        $timeout(function() {
-          WindowRef.nativeWindow.location = ctrl.TEACH_FORM_URL;
-        }, 150);
-        return false;
-      };
+  getStaticImageUrl(imagePath: string): string {
+    return this.urlInterpolationService.getStaticImageUrl(imagePath);
+  }
 
-      ctrl.$onInit = function() {
-        const hash = WindowRef.nativeWindow.location.hash.slice(1);
+  onApplyToTeachWithOppia(): boolean {
+    this.siteAnalyticsService.registerApplyToTeachWithOppiaEvent();
+    setTimeout(() => {
+      this.windowRef.nativeWindow.location.href = this.TEACH_FORM_URL;
+    }, 150);
+    return false;
+  }
+}
 
-        if (ALLOWED_TABS.includes(hash)) {
-          ctrl.activeTabName = hash;
-        }
-        WindowRef.nativeWindow.onhashchange = function() {
-          const hashChange = WindowRef.nativeWindow.location.hash.slice(1);
-          if (ALLOWED_TABS.includes(hashChange)) {
-            ctrl.activeTabName = hashChange;
-          }
-        };
-      };
-    }
-  ]
-});
+angular.module('oppia').directive(
+  'teachPage', downgradeComponent(
+    {component: TeachPageComponent}));
```

Comment:
Can this be moved to the above line if it does not exceeds line limit (80 characters)?

Reply:
Done